### PR TITLE
Add focal loss

### PR DESCRIPTION
**Pull request by davegrays**
Adds focal loss, sorry I fucked up the branch name

### CODE REVIEW

File: ultralytics/utils/loss.py [hunk 1]
```diff
@@ -36,16 +36,17 @@ def forward(pred_score, gt_score, label, alpha=0.75, gamma=2.0):
 
 
 class FocalLoss(nn.Module):
-    """Wraps focal loss around existing loss_fcn(), i.e. criteria = FocalLoss(nn.BCEWithLogitsLoss(), gamma=1.5)."""
+    """Wraps focal loss around existing loss_fcn(), i.e. criteria = FocalLoss(nn.BCEWithLogitsLoss(), gamma=2.0)."""
 
-    def __init__(self):
+    def __init__(self, reduction=None, pos_weight=None):
```
Comment (zshureih) [inline ultralytics/utils/loss.py]: Do we need the reduction parameter if we're always going to pass None/"none"

Comment (davegrays) [reply ultralytics/utils/loss.py]: good point

**Review by zshureih — APPROVED** [review]
I like the change to "class loss" and the focal loss changes make sense
